Formula/node.rb

```diff
@@ -107,12 +89,18 @@ def post_install
       # Dirs must exist first: https://github.com/Homebrew/legacy-homebrew/issues/35969
       mkdir_p HOMEBREW_PREFIX/"share/man/#{man}"
       rm_f Dir[HOMEBREW_PREFIX/"share/man/#{man}/{npm.,npm-,npmrc.,package.json.}*"]
-      ln_sf Dir[libexec/"npm/lib/node_modules/npm/man/#{man}/{npm,package.json}*"], HOMEBREW_PREFIX/"share/man/#{man}"
+      ln_sf Dir[libexec/"npm/man/#{man}/{npm,package.json}*"], HOMEBREW_PREFIX/"share/man/#{man}"
     end
 
     npm_root = node_modules/"npm"
     npmrc = npm_root/"npmrc"
     npmrc.atomic_write("prefix = #{HOMEBREW_PREFIX}\n")
+
+    # This is a minor hack that let's us keep a working npm
+    # in both HOMEBREW_PREFIX & also privately in libexec.
```

Comment:
This replaces our existing hacks around handling the fact Homebrew wants to keep two perfectly working copies of `npm` on one system, including the one tucked away privately in `libexec` so that the user can break their own `npm` without it breaking other Homebrew formulae.

Essentially, and understandably, `npm` expects you to have installed the `npm` executable in `bin`, which effectively points at `../lib/node_modules/npm/bin/npm_cli.js`. That used to be fine because we used to install `npm` with `make install` even though the benefit to users was pretty much nil over this newer proposed system. This running `make install` used to create `libexec/bin`, `libexec/etc` and `libexec/lib/node_modules/npm`.

As discussed elsewhere the `make install` logic is not much more for your basic `make install` than going _"Hey npm, install yourself with yourself"_, which since we don't bootstrap `npm` with another version of `npm` mandates the use of the new symlinks are created when `npm install` is pointed at a directory logic.

That _"use symlinks when passed files/directories rather than a tarball"_ logic broke Homebrew's existing `npm` installation dance, as well as the assumptions we had made in `brew` about `npm`'s behaviour. This fixes things on the Homebrew side without having to fake an installation or first install an older `npm` into the `buildpath` and use that for bootstrapping. The latter is theoretical, I haven't tried it because I don't dislike myself quite that much.